### DISCUSSION TITLE
Inner parts of polygons appearing black when importing Shape file

### DISCUSSION THREAD

**Discussion by AtelierCartographique** [domlysz/BlenderGIS#732] [Q&A]
Hello,

I'm having an issue when importing Shape files in Blender through BlenderGIS.

Here's the shape file in QGIS:

<img width="968" alt="Capture d’écran, le 2022-10-19 à 15 43 35" src="https://user-images.githubusercontent.com/41299938/196788659-1e2571bf-523e-400b-b224-3ef5600805d1.png">

And here's how it renders in Blender:

<img width="1234" alt="Capture d’écran, le 2022-10-19 à 15 39 33" src="https://user-images.githubusercontent.com/41299938/196788394-9b23471e-517a-450d-b964-83a9df3eeb05.png">

The inner parts are all rendered in black. Is it a known issue with BlenderGIS, and is there a way to fix this in Blender?

Thanks a lot!

Rodolphe

**Comment by atseewal** [comment]
I know I'm a bit late, but are you able to share the example shape file? I'm curious if I can replicate the issue on my machine.

**Comment by domlysz** [comment]
This is a know issue, the script can handle multipart geometry but if the part defines a polygon hole then a new face will be created. This case is difficult to handle when building the mesh. Also, according to the shapefile spec, the points of a polygon are ordered clockwise and the points of a polygon hole are ordered counterclockwise, that's why this "hole face" has a negative normal and appears black.
